code/player/TTTPlayer.Hints.cs

```diff
@@ -21,29 +21,37 @@ private void TickEntityHints()
 
             IEntityHint target = player.IsLookingAtType<IEntityHint>(MAX_HINT_DISTANCE);
 
-            //If we're looking at a valid target and we currently have a hint displayed, double check that the entity still wants us to hint.
             if (target != null && _currentHint != null)
             {
+                _currentHint.UpdateHintPanel();
+
                 if (!target.CanHint(player))
                 {
                     DeleteHint();
                     return;
                 }
             }
 
-            //If we are looking at a target and don't have a current hint, let's see if we can make one.
+            // If we are looking at a target and don't have a current hint, let's see if we can make one.
             if (target != null)
             {
                 if (target.CanHint(player) && _currentHint == null)
                 {
-                    _currentHint = target.DisplayHint(player); //Retrieves panel information from entity
+                    _currentHint = target.DisplayHint(player);
```

Comment:
There was a bug previously I fixed 
https://github.com/TTTReborn/tttreborn/commit/0af6c6fd968f56cd081e0f03e3bad11a03dbec7f

It still may occur here

Reply:
Good catch! I added your changes in.